### ISSUE TITLE
Add pagination controls to the table in the dashboard view

### ISSUE THREAD

**Issue by sergioluka** [lightdash/lightdash#8934]
### Description

 In the dashboard UI, if a table is rendered as a chart there are no pagination controls exposed.

### What problem does this solve?

We need to have data in the table without any `LIMIT`s and pagination could be a reasonable solution to that.

**Comment by TuringLovesDeathMetal** [comment]
👋 Hey @sergioluka 

I'm not too sure I understand the issue here...

WE don't actually support tables to have no limits (there is a maximum row limit of 5,000). 

Are you saying that when there are 5,000 rows, it's difficult to scroll through the results? And you'd like to have it paginated then?

**Comment by sergioluka** [comment]
@TuringLovesDeathMetal We'd like to have no limit  to the query and control the amount of the data shown in one page of the table using pagination. Does that make sense?

**Comment by sergioluka** [comment]
@TuringLovesDeathMetal I don't think that a hard limit (or any limit, to be honest) is going to be something practical for us.
It can be tricky not to know if we are seeing the total existing amount of results after all the filters are applied or not.
That is why imho pagination is a great solution as it limits the data output but also allows you to know the totals and lets you go through all the data page by page.

**Comment by jesshitchcock** [comment]
Noting that the customer above wants to see table charts paginated in embedded dashboards.

**Comment by torimwhaley** [comment]
Commenting to note that another user has requested this with the following suggestion:

The requirement would be to allow for the user experience in the "chart section" to be the same as in the "results section" (= allowing "pages or scroll", allowing pagination 1/X, allowing to display the line totals, + ideally choosing how many records you want to display per page).

For now, the UX is "full scroll" by default with no possibility of changing it, and it can cause long response time or make the browser freeze for some time if the table contains too many records.

<img width="2120" height="818" alt="Image" src="https://github.com/user-attachments/assets/e40d7c3a-805c-4992-bc2d-3b7d50376bb2" />
